compiler/rustc_trait_selection/src/traits/error_reporting/suggestions.rs

```diff
@@ -2513,6 +2521,29 @@ impl<'a, 'tcx> InferCtxtExt<'tcx> for InferCtxt<'a, 'tcx> {
             }
         }
     }
+
+    fn suggest_floating_point_literal(
+        &self,
+        obligation: &PredicateObligation<'tcx>,
+        err: &mut DiagnosticBuilder<'_>,
+        trait_ref: &ty::PolyTraitRef<'tcx>,
+    ) {
+        let rhs_span = match obligation.cause.code() {
+            ObligationCauseCode::BinOp { rhs_span: Some(span), is_lit } if *is_lit => span,
+            _ => return,
+        };
+        if let ty::Float(_) = trait_ref.skip_binder().self_ty().kind() {
+            if let ty::Infer(InferTy::IntVar(_)) = trait_ref.skip_binder().substs.type_at(1).kind()
+            {
+                err.span_suggestion_verbose(
+                    rhs_span.shrink_to_hi(),
+                    "consider using a floating-point literal by writing it with `.0`",
+                    String::from(".0"),
+                    Applicability::MaybeIncorrect,
+                );
+            }
+        }
```

Comment:
This can also be written as follows, but for this case it doesn't buy us much.

```suggestion
        match (trait_ref.skip_binder().self_ty().kind(), trait_ref.skip_binder().substs.type_at(1).kind()) {
            (ty::Float(_), ty::Infer(InferTy::IntVar(_))) => {
                err.span_suggestion_verbose(
                    rhs_span.shrink_to_hi(),
                    "consider using a floating-point literal by writing it with `.0`",
                    String::from(".0"),
                    Applicability::MaybeIncorrect,
                );
            }
            _ => {}
        }
```